ui/js/class.CodeEditable.js

```diff
@@ -0,0 +1,107 @@
+import $ from '../../index.js'
+
+
+export default class CodeEditable extends $.Component {
+    constructor(...args) {
+        super(...args)
+        this.props.default( 'lang',    'html'   )
+        this.props.default( 'code',    '<div id="1205"><h1 class="title" title="Hello World">Hello World</h1></div>')
+        this.props.default( 'theme',   'prism'  )
+        this.props.default( 'version', '1.25.0' )
+
+        const cdn      = 'https://cdnjs.cloudflare.com/ajax/libs/prism/1.25.0'
+        const cssCdn   = `${cdn}/themes/${this.props.theme}.min.css`
+        const jsCdn    = `${cdn}/prism.min.js`
+        const langCdn  = `${cdn}/components/prism-${this.props.lang}.min.js`
+
+        const urls = [cssCdn, jsCdn, langCdn]
+
+        $.Html.loadBatch(urls, this._onLoad.bind(this))
+
+        this.state.codeEdited = this.props.code
+    }
+
+    _onLoad(success) { //TODO: loading check must be global
+    	// if (success) {
+    		if (this.props.code) {
+                this.highlightCode(this.state.codeEdited)
+                this.updateLineNumbers(this.state.codeEdited)
+		    }
+	    // } else {
+	    // 	console.error('There were errors loading Prism code api')
+	    // }
+    }
+
+    highlightCode(code) {
+        const highlightedCode = Prism.highlight(
+            code,
+            Prism.languages[this.props.lang],
+            this.props.lang
+        )
+        this.codeOutput.element.innerHTML = highlightedCode
+    }
+
+    onCodeChange(event) {
```

Comment:
use hook onPropCodeChange(code) instead of onCodeChange.
it is called automatically, when you set new value to "this.props.code"